### DISCUSSION TITLE
Plans to implement Hack pipeline?

### DISCUSSION THREAD

**Discussion by sdegutis** [babel/babel#13753] [General] [answered]
The proposal recently moved to Stage 2, and I'd be interested in trying it out in real code. Are there plans to implement it, with perhaps a configurable identifier (since that's the main syntax question still left to be resolved)? If so, how can I be of some minimal help in progressing this? I have a little experience working on Babel's core for a client, so I may be able to help code.

**Comment by nicolo-ribaudo** [comment, marked as answer]
It's already implemented: https://babeljs.io/docs/en/babel-plugin-proposal-pipeline-operator

Currently it supports the `#` and `%` tokens; in Babel 7.16.0 (in 1-2 months) we'll release support for `^`.

**Reply by sdegutis** [reply]
Why does this plugin not load in the official web babel repl? It made me think that plugin was obsoleted or something.

> Plugin @babel/plugin-proposal-pipeline-operator could not be loaded

**Reply by nicolo-ribaudo** [reply]
Mhh, it's either a server error (do you have any HTTP 500 error in the console?), or because it requires some options that cannot be set using the UI.

---

Also, it looks like we forgot to move it to Stage 2. It works if you enable the "Stage 1" option in the REPL.

Do you want to open a PR to move it from
https://github.com/babel/babel/blob/8af57dbfb79dba55ca7921bfd69b289d0aad916f/packages/babel-standalone/src/preset-stage-1.ts#L31-L34
to the Stage 2 file?

**Reply by sdegutis** [reply]
Sent [PR](https://github.com/babel/babel/pull/13754/files). But I forgot to ask: Is this Hack or F# pipeline?

**Reply by sdegutis** [reply]
Ah saw that it was defaulting to 'minimal', updated PR to use 'hack'.

**Reply by sdegutis** [reply]
Only error in console is:

> repl#?browsers=defau…ssumptions=%7B%7D:1 Uncaught (in promise) undefined

Still happens when I select stage 1 (but you probably know that).
